### PR TITLE
Added Comments to Code

### PR DESCRIPTION
**Pull request by ZachCarrillo**
added comments to models and controllers

### CODE REVIEW

File: CustomerContactBook/Controllers/CustomerGroupsController.cs [hunk 1]
```diff
@@ -21,13 +21,20 @@ public CustomerGroupsController(CustomerGroupContext context)
...
+        /// </summary>
         [HttpGet]
         public async Task<ActionResult<IEnumerable<CustomerGroup>>> GetGroups()
         {
             return await _context.Groups.ToListAsync();
         }
 
         // GET: api/CustomerGroups/5
+        /// <summary>
+        /// Return all customer groups with same id
```
Comment (ysemerikov) [inline CustomerContactBook/Controllers/CustomerGroupsController.cs]: id is unique identifier of group, so there may be only one group :)

File: CustomerContactBook/Controllers/CustomerGroupsController.cs [hunk 2]
```diff
@@ -43,6 +50,11 @@ public async Task<ActionResult<CustomerGroup>> GetCustomerGroup(int id)
 
         // PUT: api/CustomerGroups/5
         // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
+        /// <summary>
+        /// creates customer group with specified id if doesnt exist already
+        /// </summary>
+        /// <param name="id">id of group</param>
```
Comment (ysemerikov) [inline CustomerContactBook/Controllers/CustomerGroupsController.cs:57]: For creation usually POST method is used and id is generated on the server side. You should pass id only for getting/changing/removing existing entity. The method below `PostCustomerGroup` looks right.

In the PUT method you should only modify existing entity; if entity with passed id doesn't exist, you should return 404 NotFound.

Comment (ysemerikov) [reply CustomerContactBook/Controllers/CustomerGroupsController.cs:57]: This comment is related to endpoint behavior, not to the comments, so feel free to fix it later. But please let me know if should I expect these changes in this PR or the next one.

**Review by ysemerikov — CHANGES_REQUESTED** [review]

File: CustomerContactBook/Controllers/CustomerGroupsController.cs (hunk 2 above)
Comment (ZachCarrillo) [reply CustomerContactBook/Controllers/CustomerGroupsController.cs:57]: I just tested the PUT method and it does return 404 not found if id doesn't exist. the summary comment for the method was just incorrect due to my misunderstanding of HTTP PUT requests. though i updated the comments in my latest commit.

**Review by ysemerikov — APPROVED** [review]

**Pull request merged**